python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -321,6 +322,23 @@ def should_reconcile(
         candidates_unit: Iterable[AssetKeyPartitionKey],
         to_reconcile: AbstractSet[AssetKeyPartitionKey],
     ) -> bool:
+        # do not attempt to materialize partitions more than one day older than the latest partition
```

Comment:
I think it's worth encapsulating this in a helper to like a lot of the other sub-logic in `should_reconcile` is